Add db-prepared-statements config option (#416)
PostgREST/postgrest-docs: releases/upcoming.rst

* Allow :ref:`s_procs_variadic`.
  |br| -- `@wolfgangwalther <https://github.com/wolfgangwalther>`_

> * Allow :ref:`connection_poolers` in transaction mode.

Reviewer: To showcase the feature better
```suggestion
* Allow :ref:`connection_poolers` such as PgBouncer in transaction pooling mode.
```

  |br| -- `@laurenceisla <https://github.com/laurenceisla>`_

* Config options for showing a full OpenAPI output regardless of the JWT role privileges and for disabling it altogether. See :ref:`openapi-mode`.
  |br| -- `@steve-chavez <https://github.com/steve-chavez>`_

* Config option for logging level. See :ref:`log-level`.
  |br| -- `@steve-chavez <https://github.com/steve-chavez>`_

> * Config option for disabling prepared statements when using connection poolers in transaction mode. See :ref:`db-prepared-statements`.

Reviewer: ```suggestion
* Config option for enabling or disabling prepared statements. See :ref:`db-prepared-statements`.
```
